flytekit/common/component_nodes.py

```diff
@@ -67,7 +67,18 @@ def __init__(self, sdk_workflow=None, sdk_launch_plan=None):
         """
         self._sdk_workflow = sdk_workflow
         self._sdk_launch_plan = sdk_launch_plan
-        super(SdkWorkflowNode, self).__init__()
+        if sdk_workflow:
+            super(SdkWorkflowNode, self).__init__(sub_workflow_ref=sdk_workflow.id)
```

Comment:
Why does it matter if we set both the values and just default them to None? easier to read isn't it?

Reply:
Again to explain
wf_id = sdk_workflow.id if sdk_workflow else None
lp = sdk_launch_plan.id if sdk_launch_plan else None
and then init

Reply:
can't dereference `None`.  The underlying constructor already has default None values, but we need to pass in the `.id` of the subworkflow/launch plan object, not the object itself.

Reply:
updated